src/vacancies/vacancies/spiders/vacancies.py

```diff
@@ -0,0 +1,304 @@
+import json
+import random
+from typing import Any, Generator, List, Optional
+
+from scrapy.http import TextResponse
+from scrapy.spiders import Spider
+
+from vacancies.items import (
+    Address,
+    AreaType,
+    Employer,
+    IdName,
+    KeySkill,
+    Metro,
+    Specialization,
+    VacancyItem,
+)
+
+
+class VacanciesSpider(Spider):
+    """
+    Парсинг вакансий с api.hh.ru.
+    """
+
+    name = "vacancies"
+    start_urls = ["https://api.hh.ru/areas"]
+
+    def parse(self, response: TextResponse, **kwargs: Any) -> Optional[Generator]:
+        """
+        Парсинг данных о регионах.
+
+        :param response: Ответ от сервера.
+        :param kwargs:
+        :return:
+        """
+
+        json_res = json.loads(response.body)
```

Comment:
How scrapy handle all the exceptions which may occur during parsing?
What happens if host drops the connection or return something different from 200 status code?

According to the docs `response.body` can contain empty byte-string (not sure if it is the case here), but `json` fails to deserialize it properly, maybe it should be handled?